content/blog/2024/10/04/2024-10-04-content-security-policy-grant.adoc

```diff
@@ -0,0 +1,47 @@
+---
+:layout: post
+:title: "Alpha Omega Foundation Content Security Policy Grant"
+:tags:
+- donation
+authors:
+- markewaite
+description: >
+  Alpha Omega Foundation funds 3 months of work on Jenkins Content Security Policy
+links:
+  discourse: https://community.jenkins.io/t/content-security-policy/nnnnn
+opengraph:
+  image: /images/post-images/2024/10/04/content-security-policy-grant.png
+---
+
+Alpha-Omega has provided a grant from for three months of full-time work to improve the link:/doc/developer/security/csp/[Jenkins implementation] of link:https://content-security-policy.com/[Content Security Policy].
+The improvements will be implemented in October, November, and December of 2024.
+link:/blog/authors/shlomomdahan/[Shlomo Dahan] is implementing the improvements with technical guidance from link:/blog/authors/basil/[Basil Crow] and project management support from link:/blog/authors/gounthar/[Bruno Verachten].
+
+link:https://alpha-omega.dev/[Alpha-Omega] is an associated project of the OpenSSF, established in February 2022, funded by Microsoft, Google, and Amazon.
+The mission of Alpha-Omega is protect society by catalyzing sustainable security improvements to the most critical open source software projects and ecosystems.
+Alpha-Omega has link:https://alpha-omega.dev/grants/grantrecipients/[funded security improvements] in major open source projects like the Linux Kernel, Rust Foundation, FreeBSD, Node.js, Ruby Central, and the Eclipse Foundation.
+We are sincerely grateful for this grant to the Jenkins project.
+
+== What is Content Security Policy?
+
+From https://content-security-policy.com/[content-security-policy.com]:
+
+> _Content-Security-Policy_ is the name of a HTTP response header that modern browsers use to enhance the security of the document (or web page). The Content-Security-Policy header allows you to restrict which resources (such as JavaScript, CSS, Images, etc.) can be loaded, and the URLs that they can be loaded from.
+
+Using Content-Security-Policy (CSP), injection attacks like cross-site scripting can be prevented.
+
+== Why a grant for Jenkins?
+
+Jenkins is the leading open source automation server.
```

Comment:
I hesitate to apply this suggestion because I copied the text from https://www.jenkins.io/ and it does not use a hyphen there.

Reply:
This would typically follow the same logic since it is an descriptor in this sentence, but I don't feel as confident saying so. We could update the jenkins.io home page so that it would be more in line with the hyphenated version, but that would be a separate issue.

Reply:
This is the logic I am using for these suggestions: https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/o/open-source